### ISSUE TITLE
Available for PHP8.0?

### ISSUE THREAD

**Issue by matt-matt1** [orisai/nette-http#1]
Hi

I am building a _Nette_ project and would like to try out your libraries.  However since I am developing in a _Debian-based OS_ on a _AARM-64 processor_, it doesn't get the latest versions.  In order to use _Nette_, I was required to upgrade to PHP8.  I tried later PHP versions (8.1, 8.2, 8.3) but only 8.0 enabled _phpmyadmin_ to function without errors.  I use this frequently to manage databases.

When I tried composer require ... it said "Package _orisai/..._ has a PHP requirement incompatible with your PHP version".  So I would like to get a version compatible with **8.0**.  Is that possible?

**Comment by mabar** [comment]
Hello, that's really weird. Could you show me the full command and error message you got?

Most of Orisai libraries are currently restricted to php: ">=7.4.0 <8.3.0". All of them except orisai/cmf are installable on PHP 8.0

**Comment by mabar** [comment]
My guess is that you have PHP version set in project to older than 7.4 or newer than 8.2 (php 8.3 will be supported in approximately a week by Orisai). Please check your `composer.json` for `require > php` and `config > platform > php` options.
If they look fine or are not present, check also the default php version with `php -v`. It is likely used by Composer and may be assumed to be your target PHP version

**Comment by matt-matt1** [comment]
Hi

I think you're right!  Nette automatically placed 'require php>=8' in composer and I think that's fine.  However my CLI has version 8.3.2 (from php -v).  So it was the opposite of what I thought - mistakenly I thought your package required a higher PHP, while is was my own system being dissatisfied.

I understand this is not what Orisai is about, but do you know how I can change the PHP version for CLI without effecting any other thing?

Thanks anyway for pointing this out I am quite new to composer.  Please dismiss this as an issue.

PS Also I am interested in the translation (i18n) module, but there's no doc.  Does that mean that you're still working on it?

**Comment by mabar** [comment]
Localization is still work in progress. I may document it, but you would have to count with BC breaks. Major part of API (translation interface in php and latte) is stable, but configuration and package name will likely change.

If you are interested in it, I may document the package soon.

If you can't wait, contributte/translation package should work just fine. Probably bit more complex, because it's integration of symfony/translation, while my package is completely independent

> do you know how I can change the PHP version for CLI without effecting any other thing

You can change default php version simply by updating alias for `php command` - `sudo update-alternatives --set php /usr/bin/php8.0`

But you shouldn't need to do that. Instead, set php version in composer.json. Composer will assume this is your target PHP version for current project.

```
	"require": {
		"php": "8.0"
	}
```

**Comment by matt-matt1** [comment]
Hi

FYI, changing the require in composed didn't work I had to use update command.  Composer is now complaining about other packages need 8.1 or higher.  So I'll just wait until I get a real computer...

Regarding translation, I looked at the contributte package...  Anyway doesn't PHP natively support gettext? Shouldn't setting a locale directory (<code>setlocale</code>) and filling those directories with language-country files work, then use <code>_('...')</code> ?

Thanks for providing more help than I expected.  Often (especially government workers) will only cater for their limited category or job description.
Although I am unable to help financially, I truly wish that you get the proper backing for the venture - that for developing nette libraries.

**Comment by mabar** [comment]
> Composer is now complaining about other packages need 8.1 or higher

Latest versions of Nette packages are already incompatible with PHP 8. But don't worry, you are not really loosing on anything new, just droping compatibility. 
What happened to you is that Composer installed newest possible version, because you were using PHP 8.3. Now your newest possible version is not the same.
Simplest will be to just delete composer.lock and all requirements from composer.json. Run composer require for each package and you will be fine with PHP 8.0

> Anyway doesn't PHP natively support gettext?

It does, but gettext is absolute pain in the ass. It does not work well with generated files (Latte), does not have parametrized messages, managing po/mo files is inpractical with multiple team members (git conflicts, unreadable files) and requires installation of gettext and individual languages on server, which is often not possible on shared hostings.

> Thanks for providing more help than I expected.

You are welcome, I am not here just for my own profit. Willing to pay of by contributing in future or just help another person once you are able to is more than good enough for me.
And don't blame people who don't do that. It's way different when you do stuff that fulfills you and just makes sense than when you do your job simply to make living.

**Comment by matt-matt1** [comment]
Hi

I revamped composer using "~8.0" since I had 8.0.30.  Most requirements were the same version, but one or two went lower.  And finally I was able to install "orisai/auth".  Now I can explore the library.  Thanks - the blockage has been rectified.

Now I feel anxious to use your translation library.  But no rush - I haven't even any other language phrases...  So far I just a wish to have my application be like nette.org <http://nette.org/>, where the viewer can select their desired language.

You have a wonderful attitude and perspective on life so you'll achieve good things irrespective of support.
